middleware/src/rpcmessages/errorcodes.go

```diff
@@ -136,9 +138,31 @@ const (
 	// ErrorDummyAuthenticationNotSuccessful is thrown if the dummy authentication is not successful.
 	ErrorDummyAuthenticationNotSuccessful ErrorCode = "DUMMY_AUTHENTICATION_NOT_SUCCESSFUL"
 
-	// ErrorDummyPasswordTooShort is thrown if the provided password is too short.
-	ErrorDummyPasswordTooShort ErrorCode = "DUMMY_CHANGEPASSWORD_TOO_SHORT"
+	// ErrorAuthenticationPasswordIncorrect is thrown if the dummy authentication is not successful.
+	ErrorAuthenticationPasswordIncorrect ErrorCode = "AUTHENTICATION_PASSWORD_INCORRECT"
+
+	// ErrorAuthenticationFailed is thrown if the dummy authentication is not successful.
+	ErrorAuthenticationFailed ErrorCode = "AUTHENTICATION_FAILED"
+
+	// ErrorAuthentcationUsernameNotFound is thrown if the given username does not exist.
+	ErrorAuthenticationUsernameNotFound ErrorCode = "AUTHENTICATION_USERNAME_NOEXIST"
```

Comment:
Should we really differentiate between a failed authentication and an authentication where the username does not exist? It helps the user when he e.g. mistypes the username, but also enables an attacker to brute force usernames. IIRC it's common practice (unless you are Facebook or Google) to not reveal when a username doesn't exist.

Reply:
Currently we only have a single "admin" user, so I'll leave this distinction for now to make it easier for us. Once we support multiple users and the adding of additional users, we can work over it again. I'll add a TODO to indicate this.